### ISSUE TITLE
associação hasMany poderia gerar um Collection fluente com where

### ISSUE THREAD

**Issue by rodolforg** [taq/torm#29]
Por várias vezes, encontro erro no meu código por tentar algo como:
```php
$biblioteca->livros->where(['formato'=>'brochura'])->next();
```
sendo `$biblioteca` um objeto com `hasMany` para classe `Livro`...

O que se obtém, contudo, é o primeiro elemento de `$biblioteca->livros`, ignorando completamente o `where`.

Seria possível implementar isso?

**Comment by taq** [comment]
Mas o next retorna o próximo.

**Comment by taq** [comment]
Ah, espera, foi mal, ignora as condições do where?

**Comment by rodolforg** [comment]
Sim, ignora. Pelo menos nas minhas três classes que tentei usar e descobri que ignora o where completamente

**Comment by taq** [comment]
É, pelo que vi aqui, parece que eu não implementei mesmo. Como faz tempo que não atualizo aqui - e nem uso o TORM mais, não tenho mais projetos em PHP - até a suíte de testes caducou com alterações no PHPUnit.

**Comment by rodolforg** [comment]
O problema é que seu ORM é muito bom... Uso desde janeiro de 2016!
Só isto que me incomoda.

Em ter, 28 de mai de 2019 às 21:45, Eustáquio Rangel <
notifications@github.com> escreveu:

**Comment by taq** [comment]
Obrigado! O duro que eu não sei se vou continuar mantendo ele, o tempo aqui está curto, eu fiz para o uso dele em um projeto mesmo que o pessoal não queria usar framework e faz muito tempo que eu não tenho contato com esse projeto, que é de um cliente. Então ficou parado. :-(
Eu teria que arrumar tempo para dar uma atualizada e colocar mais recursos, mas agora tá bem complicado. :-(

**Comment by taq** [comment]
Daria para você dar uma atualizada na suíte de testes? Ai eu poderia dar uma olhada nessa questão.

**Comment by rodolforg** [comment]
Feito: https://github.com/taq/torm/pull/30

**Comment by taq** [comment]
Valeu! Estou numa correria dando treinamento aqui, no Domingo eu dou uma olhada!
